iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/AbstractDockerTreeImpl.java

```diff
@@ -42,9 +44,14 @@ public final boolean is(Kind... kind) {
   }
 
   @Override
+  @Nullable
   public TextRange textRange() {
     if (textRange == null) {
-      List<TextRange> childRanges = children().stream().map(HasTextRange::textRange).collect(Collectors.toList());
+      List<TextRange> childRanges = children().stream().map(HasTextRange::textRange)
+        .filter(Objects::nonNull).collect(Collectors.toList());
+      if (childRanges.isEmpty()) {
+        return null;
+      }
       textRange = TextRanges.merge(childRanges);
     }
     return textRange;
```

Comment:
For empty files, a Body will have two empty list as Arg instructions and DockerImages, so it mean it will literaly have no children, which mean we need to handle such use case for textRange : possibility to return null, and filter out null range when merging.